cloud/services/disks/client.go

```diff
@@ -25,21 +25,21 @@ import (
 )
 
 // Client wraps go-sdk
-type Client interface {
+type client interface {
 	Delete(context.Context, string, string) error
 }
 
 // AzureClient contains the Azure go-sdk Client
-type AzureClient struct {
+type azureClient struct {
 	disks compute.DisksClient
 }
 
-var _ Client = &AzureClient{}
+var _ client = &azureClient{}
```

Comment:
```suggestion
var _ client = (*azureClient)(nil)
```

^ no allocation needed

Reply:
Fixed it, Thanks!